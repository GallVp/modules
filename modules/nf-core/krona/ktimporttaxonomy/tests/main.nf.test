nextflow_process {

    name "Test Process KRONA_KTIMPORTTAXONOMY"
    config "./nextflow.config"
    script "../main.nf"
    process "KRONA_KTIMPORTTAXONOMY"

    tag "modules"
    tag "modules_nfcore"
    tag "krona"
    tag "krona/ktimporttaxonomy"



    test("test_krona_ktimporttaxonomy_reads") {

        when {
            params {
                module_args = '-t 3'
            }
            process {
                """
                input[0] = [
                    [id:'test',single_end:false],// meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/test_1.kraken2.reads.txt', checkIfExists: true)
                    ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/krona_taxonomy.tab', checkIfExists: true)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.html[0][1]).text.contains('Severe acute respiratory syndrome-related coronavirus'),
                    process.out.versions
                    ).match()
                }
            )
        }
    }


    test("test_krona_ktimporttaxonomy_report") {

        when {
            params {
                module_args = '-m 3 -t 5'
            }
            process {
                """
                input[0] = [
                    [id:'test',single_end:false],// meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/test_1.kraken2.report.txt', checkIfExists: true)
                    ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/krona_taxonomy.tab', checkIfExists: true)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.html[0][1]).text.contains('Severe acute respiratory syndrome-related coronavirus'),
                    process.out.versions
                    ).match()
                }
            )
        }
    }


    test("test_krona_ktimporttaxonomy_reads -- stub") {
        options '-stub'
        when {
            params {
                module_args = '-t 3'
            }
            process {
                """
                input[0] = [
                    [id:'test',single_end:false],// meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/test_1.kraken2.reads.txt', checkIfExists: true)
                    ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/krona_taxonomy.tab', checkIfExists: true)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }


    test("test_krona_ktimporttaxonomy_report -- stub") {
        options '-stub'
        when {
            params {
                module_args = '-m 3 -t 5'
            }
            process {
                """
                input[0] = [
                    [id:'test',single_end:false],// meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/test_1.kraken2.report.txt', checkIfExists: true)
                    ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/sarscov2/metagenome/krona_taxonomy.tab', checkIfExists: true)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}