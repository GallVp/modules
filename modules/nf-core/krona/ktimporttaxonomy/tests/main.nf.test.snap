{
    "test_krona_ktimporttaxonomy_report": {
        "content": [
            true,
            [
                "versions.yml:md5,59fc89b6db8fad0aa9aa06f7437a18a7"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-25T19:59:51.434122"
    },
    "test_krona_ktimporttaxonomy_report -- stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,59fc89b6db8fad0aa9aa06f7437a18a7"
                ],
                "html": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,59fc89b6db8fad0aa9aa06f7437a18a7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-25T19:54:28.461618"
    },
    "test_krona_ktimporttaxonomy_reads": {
        "content": [
            true,
            [
                "versions.yml:md5,59fc89b6db8fad0aa9aa06f7437a18a7"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-25T19:59:46.33943"
    },
    "test_krona_ktimporttaxonomy_reads -- stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,59fc89b6db8fad0aa9aa06f7437a18a7"
                ],
                "html": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,59fc89b6db8fad0aa9aa06f7437a18a7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-25T19:54:23.95162"
    }
}